### PR TITLE
feat: log/error on mismatch of client version and supported version

### PR DESCRIPTION
**Pull request by Br1ght0ne**
Close https://github.com/FuelLabs/fuels-rs/issues/1138. Shamelessly ported from `fuels-ts`'s https://github.com/FuelLabs/fuels-ts/pull/1287.

### Checklist
- [x] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels-accounts/src/lib.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ use crate::{accounts_utils::extract_message_id, provider::Provider};
 mod accounts_utils;
 pub mod predicate;
 pub mod provider;
+pub mod supported_versions;
```
Comment (hal3e) [inline packages/fuels-accounts/src/lib.rs:31]: IMO we should not expose this outside of the crate.
```suggestion
mod supported_versions;
```

File: packages/fuels-accounts/src/supported_versions.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
+use semver::Version;
+
+pub fn get_supported_fuel_core_version() -> Version {
```
Comment (hal3e) [inline packages/fuels-accounts/src/supported_versions.rs:3]:
```suggestion
fn get_supported_fuel_core_version() -> Version {
```

File: packages/fuels-accounts/src/supported_versions.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    "0.20.6".parse().unwrap()
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct VersionCompatibility {
+    pub supported_version: Version,
+    pub is_major_supported: bool,
+    pub is_minor_supported: bool,
+    pub is_patch_supported: bool,
+}
```
Comment (hal3e) [inline packages/fuels-accounts/src/supported_versions.rs:13]:
```suggestion
#[derive(Debug, PartialEq, Eq)]
pub(crate) struct VersionCompatibility {
    pub(crate) supported_version: Version,
    pub(crate) is_major_supported: bool,
    pub(crate) is_minor_supported: bool,
    pub(crate) is_patch_supported: bool,
}
```

File: packages/fuels-accounts/src/supported_versions.rs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+pub fn check_fuel_core_version_compatibility(network_version: &Version) -> VersionCompatibility {
```
Comment (hal3e) [inline packages/fuels-accounts/src/supported_versions.rs:15]:
```suggestion
pub(crate) fn check_fuel_core_version_compatibility(network_version: &Version) -> VersionCompatibility {
```

**Review by hal3e — CHANGES_REQUESTED** [review]
Looks good. Just have a request regarding visibility

**Review by iqdecay — APPROVED** [review]
LGTM, very elegant and quick solution!

File: packages/fuels-accounts/src/supported_versions.rs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+pub(crate) fn check_fuel_core_version_compatibility(
+    network_version: &Version,
+) -> VersionCompatibility {
+    let supported_version = get_supported_fuel_core_version();
+    check_version_compatibility(network_version, &supported_version)
+}
+
+fn check_version_compatibility(
+    actual_version: &Version,
```
Comment (MujkicA) [inline packages/fuels-accounts/src/supported_versions.rs:23]: If we accept owned values here, we can move the bellow call to clone() mora towards the edge of our API. I.e. we clone when passing arguments to
`check_fuel_core_version_compatibility`

**Review by MujkicA — APPROVED** [review]
:shipit:

**Review by hal3e — APPROVED** [review]
:rocket:

**Pull request merged**